examples/distributed_tree/distributed_knn.cpp

```diff
@@ -0,0 +1,100 @@
+/****************************************************************************
+ * Copyright (c) 2017-2022 by the ArborX authors                            *
+ * All rights reserved.                                                     *
+ *                                                                          *
+ * This file is part of the ArborX library. ArborX is                       *
+ * distributed under a BSD 3-clause license. For the licensing terms see    *
+ * the LICENSE file in the top-level directory.                             *
+ *                                                                          *
+ * SPDX-License-Identifier: BSD-3-Clause                                    *
+ ****************************************************************************/
+
+#include <ArborX.hpp>
+
+#include <Kokkos_Core.hpp>
+
+#include <cstdarg>
+#include <iostream>
+#include <iterator>
+#include <vector>
+
+#include <mpi.h>
+
+using ExecutionSpace = Kokkos::DefaultExecutionSpace;
+using MemorySpace = ExecutionSpace::memory_space;
+
+namespace std
+{
+std::ostream &operator<<(std::ostream &os, ArborX::PairIndexRank const &x)
+{
+  os << "(" << x.index << ", " << x.rank << ")";
+  return os;
+}
+} // namespace std
```

Comment:
I am tempted to say replace by some print function and do an explicit loop below so we don;t showcase injecting stuff into the std nor the ArborX namespace